pkg/gitops/flux/cluster.go

```
@@ -0,0 +1,261 @@
+	return fc.gitClient.Branch(fc.branch())
+}
+
+// createRemoteRepository will create a repository in the remote git provider with the user-provided configuration
+func (fc *fluxForCluster) createRemoteRepository(ctx context.Context) error {
+	logger.V(3).Info("Remote Github repo does not exist; will create and initialize", "repo", fc.repository(), "owner", fc.owner())
+
+	opts := git.CreateRepoOpts{
+		Name:        fc.repository(),
+		Owner:       fc.owner(),
+		Description: "EKS-A cluster configuration repository",
+		Personal:    fc.personal(),
+		Privacy:     true,
+	}
+
+	logger.V(4).Info("Creating remote Github repo", "options", opts)
+	if err := fc.gitClient.CreateRepo(ctx, opts); err != nil {
+		return fmt.Errorf("creating repo: %v", err)
+	}
+
+	return nil
+}
+
+// initializeLocalRepository will git init the local repository directory, initialize a git repository.
+// it will then change branches to the branch specified in the GitOps configuration
+func (fc *fluxForCluster) initializeLocalRepository() error {
+	if err := fc.gitClient.Init(); err != nil {
+		return fmt.Errorf("initializing repository: %v", err)
+	}
+
+	// git requires at least one commit in the repo to branch from
+	if err := fc.gitClient.Commit("initializing repository"); err != nil {
+		return fmt.Errorf("committing to repository: %v", err)
+	}
+
+	if err := fc.gitClient.Branch(fc.branch()); err != nil {
+		return fmt.Errorf("creating branch: %v", err)
+	}
+	return nil
+}
+
+// validateLocalConfigPathDoesNotExist returns an exception if the cluster configuration file exists.
+// This is done so that we avoid clobbering existing cluster configurations in the user-provided git repository.
+func (fc *fluxForCluster) validateLocalConfigPathDoesNotExist() error {
+	if fc.clusterSpec.Cluster.IsSelfManaged() {
+		p := path.Join(fc.writer.Dir(), fc.path())
+		if validations.FileExists(p) {
+			return fmt.Errorf("a cluster configuration file already exists at path %s", p)
```

Just out of curiosity, would this be good user experience if we return from this point? can we store the config with a different name maybe? Also I suppose this validation can't be moved up in the code?

---

I didn't write this method but only move the code to different file. But you are right, I think we can move this validation up to preflight and fail early if local path exists. We can refactor this after this PR.